packages/contracts/contracts/voting/majority/MajorityVotingBase.sol

```diff
@@ -269,41 +293,42 @@ abstract contract MajorityVotingBase is
     }
 
     /// @inheritdoc IMajorityVoting
-    function support(uint256 _proposalId) public view virtual returns (uint256) {
+    function isSupportThresholdReached(uint256 _proposalId) public view virtual returns (bool) {
         Proposal storage proposal_ = proposals[_proposalId];
 
-        return _calculatePct(proposal_.tally.yes, proposal_.tally.yes + proposal_.tally.no);
+        return
+            (RATIO_BASE - proposal_.parameters.supportThreshold) * proposal_.tally.yes >
+            proposal_.parameters.supportThreshold * proposal_.tally.no;
     }
 
     /// @inheritdoc IMajorityVoting
-    function worstCaseSupport(uint256 _proposalId) public view virtual returns (uint256) {
+    function isSupportThresholdReachedEarly(
+        uint256 _proposalId
+    ) public view virtual returns (bool) {
         Proposal storage proposal_ = proposals[_proposalId];
 
         return
-            _calculatePct(
-                proposal_.tally.yes,
-                proposal_.tally.totalVotingPower - proposal_.tally.abstain
-            );
+            (RATIO_BASE - proposal_.parameters.supportThreshold) * proposal_.tally.yes >
+            proposal_.parameters.supportThreshold *
+                (proposal_.tally.totalVotingPower - proposal_.tally.yes - proposal_.tally.abstain);
```

Comment:
I would strongly suggest to store the intermediate calculations in easy to understand variables. These are easier to read when comparing to the formula (audit friendly).

As we did until now, I would calculate the worst case support and check it against supportThreshold and (yes+no+abs) against minVotingPower

Reply:
> As we did until now, I would calculate the worst case support and check it against supportThreshold and (yes+no+abs) against minVotingPower

This way, we cannot avoid integer division.

As explained in the PR description https://github.com/aragon/core/pull/198#issue-1537873218 and based on the previous derivation of the support criterion explained in https://github.com/aragon/core/pull/198#discussion_r1081728437, the derivation of the formula is actually not difficult

$$\begin{align*}
   (1- \texttt{supportThreshold}) \cdot N_\text{yes}
   &> \texttt{supportThreshold} \cdot (N_\text{no}+\texttt{remainingVotes})\\
   &> \texttt{supportThreshold} \cdot (N_\text{no}+N_\text{total}-(N_\text{yes}+N_\text{no}+N_\text{abstain}))\\
   &> \texttt{supportThreshold} \cdot (N_\text{total}-N_\text{yes}-N_\text{abstain})
\end{align*}$$

I wouldn't introduce more local variable as they, IMO, wouldn't bring more clarity compared to the derived formula.
If you insist, I can add `uint256 noVotesWorstCase = proposal_.tally.totalVotingPower - proposal_.tally.yes - proposal_.tally.abstain` but I think it is more confusing and people should look in the formula / docs.

Reply:
> As we did until now, I would calculate the worst case support and check it against supportThreshold and (yes+no+abs) against minVotingPower

This way, we cannot avoid integer division, so it is not possible. Moreover, we don't need `minVotingPower` for the support criterion.

As explained in the PR description https://github.com/aragon/core/pull/198#issue-1537873218 and based on the previous derivation of the support criterion explained in https://github.com/aragon/core/pull/198#discussion_r1081728437, the derivation of the formula is actually not difficult

$$\begin{align*}
   (1- \texttt{supportThreshold}) \cdot N_\text{yes}
   &> \texttt{supportThreshold} \cdot N_\text{no, worst-case}\\
   &> \texttt{supportThreshold} \cdot (N_\text{no}+\texttt{remainingVotes})\\
   &> \texttt{supportThreshold} \cdot (N_\text{no}+N_\text{total}-(N_\text{yes}+N_\text{no}+N_\text{abstain}))\\
   &> \texttt{supportThreshold} \cdot (N_\text{total}-N_\text{yes}-N_\text{abstain})
\end{align*}$$

> I would strongly suggest to store the intermediate calculations in easy to understand variables. These are easier to read when comparing to the formula (audit friendly).

I wouldn't introduce more local variable as they, IMO, wouldn't bring more clarity compared to the derived formula.
If you insist, I can add `uint256 noVotesWorstCase = proposal_.tally.totalVotingPower - proposal_.tally.yes - proposal_.tally.abstain`, but I think it is more confusing and people should look in the formula / docs.

Reply:
Thank you again @Michael-A-Heuer. Same as before <3

I would anyway recommend to have `noVotesWorstCase` as you propose, since it makes the formula easier to read as it looks in the first step of your formula expansion. 

Also, we can make sure that these steps are visible immediately before the computation, so that there is no doubt/friction about what happens :+1:

Reply:
Added `noVotesWorstCase` in 4b821d86e9d08d7d02fb403c691127a5f1b8954e